google/cloud/bigquery/_job_helpers.py

```diff
@@ -611,19 +659,33 @@ def _wait_or_cancel(
     retry: Optional[retries.Retry],
     page_size: Optional[int],
     max_results: Optional[int],
+    callback: Callable,
 ) -> table.RowIterator:
     """Wait for a job to complete and return the results.
 
     If we can't return the results within the ``wait_timeout``, try to cancel
     the job.
     """
     try:
-        return job.result(
+        query_results = job.result(
```

Comment:
TODO: send a query running (received, since it might be queued?) event before we call result.